### PR TITLE
[CLI] Add flag to disable certificate verification

### PR DESCRIPTION
**Pull request by ZacBlanco**
## Description

Adds `--insecure` flag to the CLI to skip SSL certificate verification. 

This is a cherry-pick of https://github.com/trinodb/trino/commit/1c5b9215a2d1b04b5f84e2b3e86 with modifications to allow it to compile within PrestoDB

## Motivation and Context

When deploying presto behind an SSL-enabled proxy or configuring it directly using systems like kubernetes, the system will issue self-signed certificates which can make debugging more difficult when clients have not been properly configured with the CA. When debugging clusters which are deployed this way it would be useful to be able to connect to them directly by just bypassing certificate verification.

## Impact

Additional flag for the CLI

## Test Plan

- New unit test

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== RELEASE NOTES ==

General Changes
* Add a flag to the Presto CLI which allows skipping SSL certificate verification `:pr:23780`
```

### CODE REVIEW

**Comment by steveburnett** [comment]
Nit for release note entry for consistency with the [Order of changes](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#order-of-changes) in the Release Notes Guidelines. 

```
== RELEASE NOTES ==

General Changes
* Add a flag to the Presto CLI which allows skipping SSL certificate verification `:pr:23780`
```

File: presto-docs/src/main/sphinx/clients/presto-cli.rst [hunk 1]
```diff
@@ -177,7 +192,8 @@ OPTIONS
             key=value)
 
         --runtime-stats
-            Enable runtime stats information. Flag must be used in conjunction with the --debug flag
+            Enable runtime stats information. Flag must be used in conjunction
+            with the --debug flag
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/clients/presto-cli.rst:196]:
```suggestion
            with ``--debug``
```

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for the doc, especially updating to capture other missing information that was not part of this PR! A minor nit of formatting, and otherwise looks great.

File: presto-docs/src/main/sphinx/clients/presto-cli.rst (hunk 1 above)
Comment (ZacBlanco) [reply presto-docs/src/main/sphinx/clients/presto-cli.rst:196]: Hmm, the output here is just copy-and-paste directly from the terminal into the doc. I'm not sure if making additional modifications will be fruitful as if we need to update it in the future the changes might get lost.

Is this flag not clearly readable without the backticks? I haven't actually checked the rendering in my browser

Comment (steveburnett) [reply presto-docs/src/main/sphinx/clients/presto-cli.rst:196]: Good point! Yeah, forget this nit, it's fine.

**Review by steveburnett — DISMISSED** [review]
LGTM! (docs)

Pull branch, local doc build, looks good.

**Review by imjalpreet — APPROVED** [review]
Thank you, this is really useful for development and local debugging.

**Review by elharo — APPROVED** [review]

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

**Review by hantangwangd — APPROVED** [review]

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**